crates/flux-middle/src/queries.rs

```diff
@@ -79,6 +79,10 @@ pub enum QueryErr {
         trait_id: DefId,
         name: Symbol,
     },
+    /// Need to `-Flean=check` for externally proven items
+    MissingLeanCheck {
+        def_id: DefId,
+    },
```

Comment:
We don't need this to be a `QueryErr`; it can be a "normal error" that we emit right away.